### PR TITLE
[SDK] network error leads to unwanted behavior

### PR DESCRIPTION
**Pull request by sebbi08**
# Readiness checklist

-   [ ] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

**Comment by sebbi08** [comment]
When there is a network error between SDK and connector the `errorPayload` is undefined and the SDK crashes with an `can not read property id of undefined `

File: packages/sdk/src/endpoints/Endpoint.ts [hunk 1]
```diff
@@ -47,7 +47,10 @@ export abstract class Endpoint {
 
         if (httpResponse.status !== expectedStatus) {
             const errorPayload = httpResponse.data.error;
```
Comment (jkoenig134) [inline packages/sdk/src/endpoints/Endpoint.ts:49]: when does this happen? when you call a route that does not exist on the connector?

File: packages/sdk/src/endpoints/Endpoint.ts [hunk 2]
```diff
@@ -47,7 +47,10 @@ export abstract class Endpoint {
...
-
+            if (!errorPayload) {
+                const fullUrl = `${httpResponse.request.protocol}//${httpResponse.request.getHeader("host")}${httpResponse.request.path}`;
```
Comment (jkoenig134) [inline packages/sdk/src/endpoints/Endpoint.ts:51]: I'd leave out protocol, host and only inject the path

File: packages/sdk/src/endpoints/Endpoint.ts [hunk 3]
```diff
@@ -47,7 +47,10 @@ export abstract class Endpoint {
...
+                throw new Error(`Http request to connector route ${fullUrl} failed with: ${httpResponse.status} ${httpResponse.statusText} ${httpResponse.data}`);
```
Comment (jkoenig134) [inline packages/sdk/src/endpoints/Endpoint.ts:52]:
```suggestion
                throw new Error(`Http request to connector route '${fullUrl}' failed with status '${httpResponse.status}': ${httpResponse.statusText} ${httpResponse.data}`);
```

File: packages/sdk/src/endpoints/Endpoint.ts (hunk 1 above)
Comment (jkoenig134) [reply packages/sdk/src/endpoints/Endpoint.ts:49]: oh, I didn't though about that.

File: packages/sdk/src/endpoints/Endpoint.ts (hunk 2 above)
Comment (sebbi08) [reply packages/sdk/src/endpoints/Endpoint.ts:51]: As this is mostly logged when connecting to the wrong server/port, I would leave it in there.

Comment (jkoenig134) [reply packages/sdk/src/endpoints/Endpoint.ts:51]: as this is user specified information, please leave it out.

Comment (jkoenig134) [reply packages/sdk/src/endpoints/Endpoint.ts:51]: The more I think about it I highly doubt we even need the path. The connector always responds with an error object (even for 404s).

Comment (slavistan) [reply packages/sdk/src/endpoints/Endpoint.ts:51]: If you keep the host use `.request.host`, instead of retrieving it from the request header.

**Review by jkoenig134 — APPROVED** [review]

File: packages/sdk/src/endpoints/Endpoint.ts (hunk 1 above)
Comment (slavistan) [reply packages/sdk/src/endpoints/Endpoint.ts:49]: >when you use a non connect web server

You wouldn't get an HTTP response but an exception instead in that case and `makeResult` would never execute (for example, coming from a call to `Endpoint.post()`). If you want to catch socket hangups etc. you'd have to wrap the HTTP requests in a `try/catch` block and catch `AxiosError`s. `makeResult` is only ever entered after a completed HTTP request/response cycle.

Comment (jkoenig134) [reply packages/sdk/src/endpoints/Endpoint.ts:49]: Actually it depends on how you configure axios.

Comment (slavistan) [reply packages/sdk/src/endpoints/Endpoint.ts:49]: Unless I'm mistaken, axios uses exceptions to signal errors on the network level (socket hangups, unknown hosts, timeouts, ...) and this behavior cannot be changed. So if we want the SDK to gracefully handle these errors, we have to wrap the calls to `this.httpClient.{get,post, ...}()` inside _Endpoint.ts_.

What can be configured though is to not raise an exception for non-2xx and non-3xx HTTP Response codes, which it does by default (and the SDK does override this default behavior, see [here](https://github.com/nmshd/connector/blob/f3a219d13eaae672f3d188db81809f7413bcbd9d/packages/sdk/src/ConnectorClient.ts#L38)). May you're referring to this?

Comment (jkoenig134) [reply packages/sdk/src/endpoints/Endpoint.ts:49]: Yeah, can be that I confused that. But then I am a bit confused about this PR. I don't see the usecase if network errors are already thrown errors.

Comment (sebbi08) [reply packages/sdk/src/endpoints/Endpoint.ts:49]: Hm, yea I rechecked @slavistan is right, real network errors are thrown. This code would only be run when running against a non connector web server or also a proxy that sends some error data back that is not in the connector format.

Comment (jkoenig134) [reply packages/sdk/src/endpoints/Endpoint.ts:49]: But I think this is still ok.

**Pull request merged**